source/macos/AdaptiveCards/AdaptiveCards/AdaptiveCard.swift

```diff
@@ -38,11 +38,60 @@ open class AdaptiveCard {
 }
 
 public struct RenderConfig {
-    public static let `default` = RenderConfig(isDarkMode: false)
-    
+    public static let `default` = RenderConfig(isDarkMode: false, buttonConfig: .default)
     let isDarkMode: Bool
+    let buttonConfig: ButtonConfig
     
-    public init(isDarkMode: Bool) {
+    public init(isDarkMode: Bool, buttonConfig: ButtonConfig) {
         self.isDarkMode = isDarkMode
+        self.buttonConfig = buttonConfig
+    }
+}
+
+public struct ButtonColorConfig {
+    public static let `default` = ButtonColorConfig()
+    // backgroundColor
+    let backgroundColor: NSColor
+    
+    // buttonColor
+    let buttonColor: NSColor
+    let activeButtonColor: NSColor
+    let hoverButtonColor: NSColor
+    
+    // textColor
+    let textColor: NSColor
+    let activeTextColor: NSColor
+    
+    // borderColor
+    let borderColor: NSColor
+    let activeBorderColor: NSColor
+        
+    public init(backgroundColor: NSColor = .clear, buttonColor: NSColor = .clear, activeButtonColor: NSColor = .clear, hoverButtonColor: NSColor = .clear, textColor: NSColor = .clear, activeTextColor: NSColor = .clear, borderColor: NSColor = .clear, activeBorderColor: NSColor = .clear) {
```

Comment:
Guess we can ignore having `default` values for parameters here as it's kindof misleading.

Reply:
adding some color to make a default more generic button..will that be fine?

Reply:
No Rohit. We can remove all the `.clear` values here. Let all params be required.
In `ButtonConfig.default` we can have genric values for different button types.